hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/tasks/FileSizeCountTaskHelper.java

```diff
@@ -43,14 +47,26 @@ public abstract class FileSizeCountTaskHelper {
   // Static lock object for table truncation synchronization
   private static final Object TRUNCATE_LOCK = new Object();
 
+  /**
+   * GLOBAL lock for cross-task synchronization during DB writes.
+   * 
+   * Scope: Shared across ALL tasks (FSO, OBS, Legacy)
+   * Protects: RocksDB read-modify-write operations
+   * Purpose: Ensures atomic updates when multiple tasks flush concurrently to the same bins
+   * 
+   * IMPORTANT: Callers MUST acquire this lock before calling writeCountsToDB().
+   * This lock should NOT be acquired inside writeCountsToDB() to avoid nested locking.
+   */
+  public static final ReentrantReadWriteLock FILE_COUNT_WRITE_LOCK = 
```

Comment:
Now we have so many locks local, global etc,  Document the lock hierarchy clearly in code comments. Also Add lock ordering diagram in PR description.